File: Ruby/kyu-7/_0035__sum-of-the-first-nth-term-of-series/solution.rb
def series_sum(n)
  res, step, count = 0, 3, 1

  n.times do
    res += 1.0 / count
    count += step
  end

  "%.2f" % res
end

=begin
or
def series_sum(n)
  '%.2f' % (0...n).inject(0){ |sum, i| sum + 1.0/(1+i*3) }
end
=end

File: JavaScript/kyu-7/_0035__sum-of-the-first-nth-term-of-series/solution.js
function SeriesSum(n) {
  let res = 1;
  const step = 3;
  let divider = 1;
  while (n > 1) {
    res += 1 / (divider += step);
    n--;
  }

  return n === 0 ? '0.00' : res.toFixed(2);
}

/*
or 
function SeriesSum(n) {
  for (var s = 0, i = 0; i < n; i++) {
    s += 1 / (1 + i * 3)
  }
  
  return s.toFixed(2)
}

or
function SeriesSum(n, s = 0) {
    return n ? SeriesSum(n - 1, s + 1 / (3 * n - 2)) : s.toFixed(2)
}
*/
